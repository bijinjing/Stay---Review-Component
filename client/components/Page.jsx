import React from 'react';


class Page extends React.Component{
  constructor(props) {
    super(props)

    this.state={
      currentPage:1,
      pageHead:1
    }

  }


  render(){
    return(
      <form>
          {this.state.currentPage !== 1 && <input id="left" value="left" type="button"/>}
          <input value='1' type="button"/>
          {[1,2,3].includes(this.state.currentPage) && <input value={this.state.pageHead} type="button"/>}
          {[2,3,4].includes(this.state.currentPage) && <input value={this.state.pageHead+1} type="button"/>}
          <input value={this.state.pageHead+2} type="button"/>
          <input value={this.props.lastPage-2} type="button"/>
          <input value={this.props.lastPage-1} type="button"/>

          <input value={this.props.lastPage} type="button"/>
          {this.state.currentPage !== this.props.lastPage && <input id="right" value="right" type="button"/>}
      </form>
    )
  }
}

export default Page


