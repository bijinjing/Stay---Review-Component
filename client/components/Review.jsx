import React from 'react';
import ReadMore from './ReadMore.jsx';
import moment from 'moment';
import styled from 'styled-components'

const Review_profile = styled.div`
  box-sizing: border-box;
  border-bottom: 1px ridge lightgrey
`;

const Review_content = styled.div`
  margin-bottom:30px;
`;

const Profile = styled.div`
  display:inline-flex;
  align-items: flex-start;
  margin-top:20px;
  margin-bottom:17px
`;

const ProfileTitle = styled.div`
  display:inline-block;
  margin-left:20px;
`;

const ProfileImg = styled.img`
  display:inline-block;
  height:48px;
  border-Radius:50%
`
const Review = (props) => {
  return(
    <Review_profile>
      <Profile>
        <ProfileImg src={props.data.profile}></ProfileImg>
        <ProfileTitle>
          <div style={{fontWeight:`bold`,marginBottom:'3px'}}>{props.data.name}</div>
          <div>{moment(props.data.review_date).format('MMMM YYYY')}</div>
        </ProfileTitle>
      </Profile>
      <Review_content>
        {!props.searchStatus && props.data.comments.split(" ").length > 30 ? <ReadMore comments = {props.data.comments}/> : <div>{props.data.comments}</div>}
      </Review_content>
    </Review_profile>

  )
}


export default Review



