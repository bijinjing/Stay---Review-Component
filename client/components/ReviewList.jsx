import React from 'react'
import Review from './Review.jsx'

class ReviewList extends React.Component{
  constructor(props) {
    super(props)

  }

  render(){
    return(
      <div>
        {this.props.reviewList.map((data) => { return <Review data={data}/>})}
      </div>
    )
  }
}

export default ReviewList