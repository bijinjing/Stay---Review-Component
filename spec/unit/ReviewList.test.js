import React from 'react';
import Review from '../../client/components/Review.jsx';
import Page from '../../client/components/Page.jsx';
import ReviewList from '../../client/components/ReviewList.jsx';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

describe('ReviewList', ()=>{
  it('renders without crashing when only one page', () => {
    const props = {
      reviewList:[
        {
          profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg',
          name:"jinjing",
          review_data:"3/14/19",
          comments:"it is good"
        },
        {
          profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/129.jpg',
          name:"derek",
          review_data:"3/20/19",
          comments:"great"
        }
      ],
      searchStatus:false,
      currentPage:1,
      lastPage:1,
      renderCurrentPage:() => {},
      changeCurrentPage:() => {}
    }

    const wrapper = shallow(<ReviewList {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without crashing when more than one page', () => {
    const props = {
      reviewList:[
        {
          profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg',
          name:"jinjing",
          review_data:"3/14/19",
          comments:"it is good"
        },
        {
          profile:'https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/129.jpg',
          name:"derek",
          review_data:"3/20/19",
          comments:"great"
        }
      ],
      searchStatus:false,
      currentPage:1,
      lastPage:2,
      renderCurrentPage:() => {},
      changeCurrentPage:() => {}
    }

    const wrapper = shallow(<ReviewList {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})