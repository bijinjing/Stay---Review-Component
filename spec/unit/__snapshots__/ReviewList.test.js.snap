// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList renders without crashing when more than one page 1`] = `
<div>
  <Review
    data={
      Object {
        "comments": "it is good",
        "name": "jinjing",
        "profile": "https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg",
        "review_data": "3/14/19",
      }
    }
  />
  <Review
    data={
      Object {
        "comments": "great",
        "name": "derek",
        "profile": "https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/129.jpg",
        "review_data": "3/20/19",
      }
    }
  />
  <Page
    changeCurrentPage={[Function]}
    currentPage={1}
    lastPage={2}
    renderCurrentPage={[Function]}
    searchStatus={false}
  />
</div>
`;

exports[`ReviewList renders without crashing when only one page 1`] = `
<div>
  <Review
    data={
      Object {
        "comments": "it is good",
        "name": "jinjing",
        "profile": "https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/128.jpg",
        "review_data": "3/14/19",
      }
    }
  />
  <Review
    data={
      Object {
        "comments": "great",
        "name": "derek",
        "profile": "https://s3.amazonaws.com/uifaces/faces/twitter/maxlinderman/129.jpg",
        "review_data": "3/20/19",
      }
    }
  />
</div>
`;
